### PR TITLE
Update developers guide use cases

### PR DESCRIPTION
**Pull request by kxyoke**
Addresses #8

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -544,6 +544,58 @@ Use case resumes at step 1.
 +
 Use case resumes at step 1.
 
+[discrete]
+=== Use case: UC09: Browse
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]: Any possible extensions here? E.g. no Internet access

Also, a suggestion for the use case title:
```suggestion
=== Use case: UC09 - Browse the web
```

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -362,7 +362,7 @@ _{More to be added}_
 (For all use cases below, the *System* is the `Mark` and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case 01: List bookmarks
+=== Use case UC01: List bookmarks
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]: Suggested change in format to match the textbook (and likewise for the other UCs):
```suggestion
=== Use case: UC01 - List bookmarks
```

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -372,7 +372,7 @@ _{More to be added}_
 Use case ends.
 
 [discrete]
-=== Use case 02: Add bookmark
+=== Use case UC02: Add bookmark
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]:
```suggestion
=== Use case: UC02 - Add bookmark
```

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -391,7 +391,7 @@ Use case ends.
 Use case resumes at step 2.
 
 [discrete]
-=== Use case 03: Delete bookmark
+=== Use case UC03: Delete bookmark
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]:
```suggestion
=== Use case: UC03 - Delete bookmark
```

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -544,6 +544,58 @@ Use case resumes at step 1.
 +
 Use case resumes at step 1.
 
+[discrete]
+=== Use case: UC09: Browse
+*MSS*
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]: **MSS** is not showing up as a heading on the Netlify preview (likewise for UC10 and UC11).
```suggestion
*MSS*

```

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -544,6 +544,58 @@ Use case resumes at step 1.
...
+1. User requests to browse the web.
+2. Mark provides in-built web browser.
+3. User browses the web.
++
+Use case ends.
+
+[discrete]
+=== Use case: UC10: Add bookmarks from web browser
+*MSS*
+1. User [.underline]#browses the web (UC09)#.
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]: Can we make this a link?

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -372,7 +372,7 @@ _{More to be added}_
 Use case ends.
 
 [discrete]
-=== Use case 02: Add bookmark
+=== Use case: UC02 - Add bookmark
```
Comment (Na-Nazhou) [inline docs/DeveloperGuide.adoc]: Should we change this? (same for other use similar cases)
```suggestion
=== Use case: UC02 - Add bookmarks
```

Comment (dorcastan) [reply docs/DeveloperGuide.adoc]: Hm it's ok to me! Since we are only adding one bookmark in this use case.

File: docs/DeveloperGuide.adoc [hunk 8]
```diff
@@ -583,6 +593,74 @@ Mark shows an error message.
...
+1. User requests to browse the web.
+2. Mark provides in-built web browser.
+3. User browses the web.
++
+Use case ends.
+
+*Extensions*
+[none]
+* 1a. Mark does not have internet access.
+** 1a1. Mark tells user that there is no internet access and suggests user to see cache.
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]:
```suggestion
** 1a1. Mark tells user that there is no internet access and suggests that user see cache.
```

File: docs/DeveloperGuide.adoc [hunk 9]
```diff
@@ -583,6 +593,74 @@ Mark shows an error message.
...
+Use case ends.
+
+[#UC12]
+[discrete]
+=== Use case: UC12 - Add bookmarks from web browser
+
+*MSS*
+
+1. User <<UC11,[.underline]#browses the web (UC11)#>>.
+2. User requests to bookmark currently showing webpage.
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]:
```suggestion
2. User requests to bookmark the currently showing webpage.
```

File: docs/DeveloperGuide.adoc [hunk 10]
```diff
@@ -583,6 +593,74 @@ Mark shows an error message.
...
++
+Use case ends.
+
+[#UC12]
+[discrete]
+=== Use case: UC12 - Add bookmarks from web browser
+
+*MSS*
+
+1. User <<UC11,[.underline]#browses the web (UC11)#>>.
```
Comment (dorcastan) [inline docs/DeveloperGuide.adoc]: Just a note: the reference UC11 doesn't link to anything at the moment. (I checked the HTML source, and the id of the UC11 header is still "use-case-uc11-browse-the-web".) I think we can fix it in future versions.

Comment (kxyoke) [reply docs/DeveloperGuide.adoc]: Clicking on the link works on the developer's guide web page though; which page's html are you referring to?

Comment (dorcastan) [reply docs/DeveloperGuide.adoc]: Does the position of the page change? Mine goes to [this page](https://deploy-preview-47--silly-jennings-9dd59a.netlify.com/developerguide#UC11) but the fragment `#UC11` doesn't link to any header.

**Pull request merged**